lib/SILOptimizer/Transforms/CSE.cpp

```diff
@@ -182,53 +196,70 @@ class HashVisitor : public SILInstructionVisitor<HashVisitor, llvm::hash_code> {
   hash_code visitStructInst(StructInst *X) {
     // This is safe since we are hashing the operands using the actual pointer
     // values of the values being used by the operand.
-    OperandValueArrayRef Operands(X->getAllOperands());
-    return llvm::hash_combine(X->getKind(), X->getStructDecl(),
-      llvm::hash_combine_range(Operands.begin(), Operands.end()));
+    auto OperandValues =
+        X->getFunction()->hasOwnership()
+            ? X->getOperandValues(false, lookThroughOwnershipInsts)
```

Comment:
Why not create a local transform kind here? Why should the needs of CSE impact all use sites of operand values?